mailbox/store/src/main/java/org/apache/james/mailbox/store/event/MailboxUpdateParentAsLookupListener.java

```diff
@@ -0,0 +1,87 @@
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.mailbox.store.event;
+
+import java.util.List;
+
+import org.apache.james.mailbox.MailboxListener;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.exception.MailboxException;
+import org.apache.james.mailbox.model.MailboxACL;
+import org.apache.james.mailbox.model.MailboxACL.EntryKey;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.store.MailboxSessionMapperFactory;
+import org.apache.james.mailbox.store.mail.MailboxMapper;
+import org.apache.james.mailbox.store.mail.model.Mailbox;
+import org.apache.james.mailbox.store.transaction.Mapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class MailboxUpdateParentAsLookupListener implements MailboxListener {
+    private static final Logger logger = LoggerFactory.getLogger(MailboxUpdateParentAsLookupListener.class);
+    private MailboxSessionMapperFactory mailboxSessionMapperFactory;
+
+    public MailboxUpdateParentAsLookupListener(MailboxSessionMapperFactory mailboxSessionMapperFactory) {
+        this.mailboxSessionMapperFactory = mailboxSessionMapperFactory;
+    }
+    @Override
+    public ListenerType getType() {
+        return ListenerType.EACH_NODE;
+    }
+
+    @Override
+    public ExecutionMode getExecutionMode() {
+        return ExecutionMode.SYNCHRONOUS;
+    }
+
+    @Override
+    public void event(Event event) {
+        if (event instanceof EventFactory.MailboxLookupRightUpdate) {
+            try {
+                EventFactory.MailboxLookupRightUpdate mailboxLookupRightUpdate = (EventFactory.MailboxLookupRightUpdate)event;
+
+                MailboxSession mailboxSession = mailboxLookupRightUpdate.getSession();
+                MailboxPath mailboxPath = mailboxLookupRightUpdate.getMailboxPath();
+
+                MailboxMapper mapper = mailboxSessionMapperFactory.getMailboxMapper(mailboxSession);
+                List<MailboxPath> mailboxPaths = mailboxPath.getHierarchyLevels(mailboxSession.getPathDelimiter());
+                mailboxPaths.stream()
+                    .filter(hierarchyMailboxPath -> !hierarchyMailboxPath.equals(mailboxPath))
+                    .forEach(parentMailboxPath -> {
+                        updateLookupRight(mapper, parentMailboxPath, mailboxLookupRightUpdate.getEntryKey());
+                    });
+            } catch (MailboxException e) {
+                logger.error("Unable to look up MailboxMapper", e);
```

Comment:
Euh... That log message could be improved... Think of the poor **integration manager** who will try to give them some sense. What information does he need to know to evaluate and fix the problem?